Bad conditional in vSphereLogin function

robdaemon opened the pull request:
```release-note
Fixes NotAuthenticated errors that appear in the kubelet and kube-controller-manager due to never logging in to vSphere
```

With this conditional being == instead of !=, a login would never actually be attempted by this provider, and disk attachments would fail with a NotAuthenticated error from vSphere.

<!-- Reviewable:start -->

---
This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36169)
<!-- Reviewable:end -->

Commit:
Bad conditional in vSphereLogin function

With this conditional being == instead of !=, a login would never actually be attempted by this provider, and disk attachments would fail.

robdaemon commented:
I'm pretty sure HPE has a signed CLA with the Linux Foundation?

grahamhayes commented:
@k8s-bot ok to test

robdaemon commented:
There, Linux Foundation CLA signed

robdaemon commented:
What else is needed to get this PR through? I'm not sure how anyone is using Kubernetes on vSphere right now :)

grahamhayes commented:
@k8s-bot approvers

abrarshivani commented on pkg/cloudprovider/providers/vsphere/vsphere.go:
```diff
@@ -360,7 +360,7 @@ func vSphereLogin(vs *VSphere, ctx context.Context) error {
 	// retrieve client's current session
 	u, err := m.UserSession(ctx)
-	if err == nil && u == nil {
+	if err == nil && u != nil {
```
@robdaemon Thanks for doing this. This will create user session even if it fails to obtain the existing one. So, I would suggest you to do like following,
```
. . .

if err != nil {
	glog.Errorf("Error while obtaining user session")
	return err
}

if u != nil {
	return nil
}

. . . 
```
This will handle the error. You can check conditions [here](https://github.com/vmware/govmomi/blob/7e1145231d2385c8a170c8a4a5ca11e1c1b706bb/session/manager.go#L138).

robdaemon replied:
Okay, I've made the changes you request here.

Commit:
Applying changes per PR feedback -

Check for error conditions from the vSphere API and return the err if one occurs. The vSphere API does not return an err for unauthenticated users, it just returns a nil user object.

abrarshivani commented:
LGTM

kerneltime commented:
@k8s-bot ok to test

kerneltime commented:
@mikedanese Can this be merged in once the tests pass?

mikedanese commented:
/lgtm
